src/lib/y2storage/storage_manager.rb

```diff
@@ -228,6 +228,20 @@ def staging
       end
     end
 
+    # Staging devicegraph
+    #
+    # @raise [Yast::AbortException] when {#probe} returns false
+    # @see #staging
+    #
+    # @return [Devicegraph]
+    def staging!
+      success = probed? ? true : probe
+
+      raise Yast::AbortException unless success
```

Comment:
TBH, raising a `Yast::AbortException` is unexpected for me. Such an exception should be raised as the result of a user action (e.g., when the user clicks abort) or as result of certain action that makes YaST abort. But I would say that calling `StorageManager#staging!` is not the case. That is a decision the caller should take. I would prefer to raise some kind of storage exception, for example, `Y2Storage::Error` or `Y2Storage::ProbeError`.

Reply:
For example, [here](https://github.com/yast/yast-storage-ng/blob/master/src/lib/y2storage/storage_manager.rb#L612) a `Yast::AbortException` is raised only if the user decides to not recover from a `Storage::LockException`.

Reply:
I see your point. The thing is that one of the reasons the code reach that lines is precisely because the user decided to abort (`#probe` -> `#manage_probing_issues`).

But sure, we can raise a `Y2Storage::Error` and raise (or not) a `Yast::AbortException` in the callers.

Reply:
As agreed in an offline conversation,

* changed where the exception is raised
* added #probe!
* adapted #staging!